app/users/views.py

```diff
@@ -96,52 +100,40 @@ def change_password(request):
             user = form.save()
             update_session_auth_hash(request, user)  # Important!
-            messages.success(request, "Your password was successfully updated!")
+            messages.success(request, _("Your password was successfully updated!"))
             return redirect("change_password")
         else:
-            messages.error(request, "Please check errors and resubmit!")
+            messages.error(request, _("Please check errors and resubmit!"))
     else:
         form = PasswordChangeForm(request.user)
     return render(request, "registration/change_password.html", {"form": form})
 
 
-def password_reset_request(request):
-    if request.method == "POST":
-        password_reset_form = PasswordResetForm(request.POST)
-        if password_reset_form.is_valid():
-            data = password_reset_form.cleaned_data["email"]
-            associated_users = CustomUser.objects.filter(Q(email=data))
-            if associated_users.exists():
-                for user in associated_users:
-                    subject = "Password Reset Requested"
-                    email_template_name = "registration/password_reset_email.txt"
-                    c = {
-                        "email": user.email,
-                        "domain": EMAIL_HOST,
-                        "site_name": "open_plan",
-                        "uid": urlsafe_base64_encode(force_bytes(user.pk)),
-                        "user": user,
-                        "token": default_token_generator.make_token(user),
-                        "protocol": "http",
-                    }
-                    email = render_to_string(email_template_name, c)
-                    try:
-                        send_mail(
-                            subject,
-                            email,
-                            DEFAULT_FROM_EMAIL,
-                            [user.email],
-                            fail_silently=False,
-                        )
-                    except BadHeaderError:
-                        return HttpResponse("Invalid header found.")
-                    return redirect("/password_reset/done/")
-    password_reset_form = PasswordResetForm()
-    return render(
-        request=request,
-        template_name="registration/password_reset_form.html",
-        context={"password_reset_form": password_reset_form},
-    )
+class ExchangePasswordResetForm(PasswordResetForm):
+    def send_mail(
+        self,
+        subject_template_name,
+        email_template_name,
+        context,
+        from_email,
+        to_email,
+        html_email_template_name=None,
+    ):
+        """
+        Send a django.core.mail.EmailMultiAlternatives to `to_email`.
+        """
+        subject = render_to_string(subject_template_name, context)
+        # Email subject *must not* contain newlines
+        subject = "".join(subject.splitlines())
+        # Force the https for the production
+        if settings.DEBUG is False:
```

Comment:
@4lm - I am not sure this is the best practice, at least it works (the online dev server version is updated)

Reply:
I left it HTTP for all cases intentionally because HTTPS is not really needed here. Because HTTP works in all cases (dev and production). In production, the web server should forward all HTTP requests to HTTPS. It's a link and, therefore, a GET request, which means the token info is in the URL and public with both protocols (which is IMO totally OK from a security standpoint).

But I'm totally OK with your changes. You went the extra mile. OK, for me. Concerning your question ("I am not sure this is the best practice"), a dedicated flag (e.g., `USE_HTTPS`) would be more evident. Still, I think using `DEBUG` for this is also OK. I would leave your changes like they are.

Reply:
I had the issue that in the dev server when I cliked on the http link it would not lead anywhere. But if I copied it in the browser and added the `s`of https, then it was working.

Locally if I force https, then it does not work, only http works

I will leave it as is then